src/Native/Runtime/startup.cpp

```diff
@@ -39,13 +39,18 @@ UInt32 _fls_index = FLS_OUT_OF_INDEXES;
 
 
 Int32 __stdcall RhpVectoredExceptionHandler(PEXCEPTION_POINTERS pExPtrs);
-void __stdcall FiberDetach(void* lpFlsData);
+void __stdcall ThreadShutdown();
 void CheckForPalFallback();
 
 extern RhConfig * g_pRhConfig;
 
 bool InitDLL(HANDLE hPalInstance)
 {
+    if (!PalInit())
```

Comment:
Calling PalInit here will mean that it gets called twice in the PN runtime, which calls PalInit from DllMain.  The reason for PN to do it this way was that we considered the runtime to be three separate parts:  unmanaged (C++), managed (C#), and PAL.  DllMain was responsible for initializing all 3 in the proper order.  Putting the PalInit here will conflate the C++ and PAL pieces together and require some corresponding PN-side changes.  Could we call PalInit from __initialize_runtime instead?